### PR TITLE
Implement client-side UI update dbus signal

### PR DESCRIPTION
**Pull request by nntae**
in preparation for fbterm support

### CODE REVIEW

**Comment by wengxt** [comment]
So at least, can you try to make it merge-able (from web ui side)?.. I don't want to review my changes and your changes together.

Multiple ways to do it:
1. cherry-pick you commit one by one on to new master
2. reset and squash all your change into one commit.
3. rebase if it's easy to do so.

**Comment by nntae** [comment]
updated to resolve conflicts

File: src/lib/fcitx/inputcontext.cpp [hunk 1]
```diff
@@ -256,7 +256,11 @@ void InputContext::updatePreedit() {
 void InputContext::updateUserInterface(UserInterfaceComponent component,
                                        bool immediate) {
     FCITX_D();
-    d->emplaceEvent<InputContextUpdateUIEvent>(component, this, immediate);
```
Comment (wengxt) [inline src/lib/fcitx/inputcontext.cpp:259]: I think we should move this part of the code to UserInterfaceManager::flush. Thus the event is always generated.

File: src/frontend/dbusfrontend/dbusfrontend.cpp [hunk 2]
```diff
@@ -23,6 +23,17 @@
 
 namespace fcitx {
 
+namespace {
+
+void buildFormattedTextVector(
```
Comment (wengxt) [inline src/frontend/dbusfrontend/dbusfrontend.cpp:28]: two choice about the naming:
1. just construct and return the vector
2. rename from build to append (if same vector is used by this function twice).

**Pull request merged**